### PR TITLE
Cancel a stalled standard-input writer when the child process exits

### PR DESCRIPTION
**Pull request by iCharlesHu**
When `run()` feeds standard input from a source that can stall and the child process exits before consuming that input, `run()` could hang indefinitely. The input-writing task was parked awaiting its source rather than blocked inside a pipe write, so the existing `cancelAsyncIO` path (which only unblocks in-flight pipe I/O) could not release it, and the I/O task group never finished draining. Spawn a `waitForProcessTermination` waiter alongside the input-writing task that cancels the I/O group once the child exits.

Also add edge-case coverage for process I/O and AsyncIO, ported from `swift-async-process` and `swift-nio` and replace `FileDescriptor.close` with `_safelyClosed`.

### CODE REVIEW

File: Tests/SubprocessTests/IntegrationTests.swift [hunk 1]
```diff
@@ -3527,6 +3527,170 @@ extension SubprocessIntegrationTests {
...
+        let result = try await _run(
+            setup,
+            input: .fileDescriptor(inputFD, closeAfterSpawningProcess: false),
+            output: .string(limit: 256),
+            error: .discarded
+        )
+        #expect(result.terminationStatus.isSuccess)
+        #expect(result.standardOutput?.trimmingNewLineAndQuotes() == content)
+        // The parent retained ownership, so closing here must succeed. If
+        // Subprocess had wrongly closed it, this would throw `.badFileDescriptor`.
```
Comment (jakepetroules) [inline Tests/SubprocessTests/IntegrationTests.swift]: well, it _might_ throw badFileDescriptor. Remember, there's no guarantee of anything if the owner of the file descriptor did not close it. It could have since been reopened and that fd# pointed to something else.

Comment (jakepetroules) [reply Tests/SubprocessTests/IntegrationTests.swift]: Also, suggest using closeAfter

**Review by jakepetroules — APPROVED** [review]
